SofaKernel/modules/SofaEigen2Solver/CMakeLists.txt

```diff
@@ -23,8 +23,6 @@ find_package(Eigen3 REQUIRED)
 
 if (SOFA_OPENMP AND "${Eigen3_VERSION}" VERSION_LESS 3.2.9)
     sofa_find_package(OpenMP BOTH_SCOPES) # will set/update SOFAEIGEN2SOLVER_HAVE_OPENMP
-else()
-    sofa_set_01(SOFAEIGEN2SOLVER_HAVE_OPENMP VALUE FALSE BOTH_SCOPES)
 endif()
```

Comment:
Without the `else`, if you decide to disable `SOFA_OPENMP` after having it enabled, you will still have `SOFAEIGEN2SOLVER_HAVE_OPENMP = 1` in CMake cache.

Reply:
Yeah, and it will do the inverse of what it should, since `#ifdef` are used

Reply:
Sorry, I've misread you.

`SOFAEIGEN2SOLVER_HAVE_OPENMP` isn't a cmake cache variable. It is unset at the beginning of each cmake call. I've tested it:

```sh
$ sed -i 's/SOFA_OPENMP:BOOL=ON/SOFA_OPENMP:BOOL=OFF/' CMakeCache.txt
$ cmake .
$ grep SOFAEIGEN2SOLVER_HAVE_OPENMP include/SofaCommon/config.h
/* #undef SOFAEIGEN2SOLVER_HAVE_OPENMP */
$ sed -i 's/SOFA_OPENMP:BOOL=OFF/SOFA_OPENMP:BOOL=ON/' CMakeCache.txt
$ cmake .
$ grep SOFAEIGEN2SOLVER_HAVE_OPENMP include/SofaCommon/config.h
#define SOFAEIGEN2SOLVER_HAVE_OPENMP
$ sed -i 's/SOFA_OPENMP:BOOL=ON/SOFA_OPENMP:BOOL=OFF/' CMakeCache.txt
$ cmake .
$ grep SOFAEIGEN2SOLVER_HAVE_OPENMP include/SofaCommon/config.h
/* #undef SOFAEIGEN2SOLVER_HAVE_OPENMP */
```

Reply:
Indeed, thanks for the tests :+1:
What happens if `#cmakedefine01` is used instead?

Reply:
OFF : `#define SOFAEIGEN2SOLVER_HAVE_OPENMP 0`
ON :  `#define SOFAEIGEN2SOLVER_HAVE_OPENMP 1`
OFF : `#define SOFAEIGEN2SOLVER_HAVE_OPENMP 0`

and in cpp files,  `#ifdef SOFAEIGEN2SOLVER_HAVE_OPENMP` will always be true since it is always defined

Reply:
Yes because the idea behind `#cmakedefine01` is to permit the usage of `#if` and `if()`. `#ifdef` surely becomes useless though.